apps/gallery/test/marionette/lib/gallery.js

```diff
@@ -240,6 +256,66 @@ Gallery.prototype = {
   },
 
   /**
+   * @return {Marionette.Element} Done Button to finish crop and pick image.
+   */
+  get cropDoneButton() {
+    return this.client.helper.waitForElement(Gallery.Selector.cropDoneButton);
+  },
+
+  /**
+   * @return {Marionette.Element} edit crop canvas showing crop overlay.
+   */
+  get editCropCanvas() {
+    return this.client.helper.waitForElement(Gallery.Selector.editCropCanvas);
+  },
+
+   /**
+   * @return {Marionette.Element} element to display image opened using
+   * gallery app open activity.
+   */
+  get openActivityImage() {
+    return this.client.helper.waitForElement(Gallery.Selector.open.image);
+  },
+
+  /**
+   * @return {Marionette.Element} header element showing file name opened using
+   * gallery app open activity.
+   */
+  get openActivityImageTitle() {
+    return this.client.helper.waitForElement(Gallery.Selector.open.title);
+  },
+
+  /**
+   * @return {Marionette.Element} save button that saves image opened using
+   * gallery app open activity.
+   */
+  get openActivitySaveButton() {
+    return this.client.helper.waitForElement(Gallery.Selector.open.saveButton);
+  },
+
+  /**
+   * @return {Marionette.Element} that shows menu options on using web activity.
+   */
+  get actionMenu() {
+    // Switch to the system app first.
+    client.switchToFrame();
+    return this.client.helper.waitForElement(Gallery.Selector.actionMenu);
+  },
+
+  /**
+   * @return {Marionette.Element} menu option that has same text as appName.
+   */
+  menuOptionButton: function(appName) {
+    var options = this.actionMenu.findElements('button');
+    for (var i = 0; i < options.length; i++) {
+      var menuOption = options[i];
+      if (menuOption.text() === appName) {
+        return menuOption;
+      }
+    }
+  },
```

Comment:
Consider putting both the actionMenu getter and the menuOptionButton() function into a separate library that can be used by both this file and fakeactivitycaller.js